src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      let item = this[i];
```

Comment:
Instead of checking if the item is null or undefined, you could check if it is not undefined. This will handle all falsy values except for the number 0 and empty strings, which should be included in the joined string.